platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/status/StatusStateMachine.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.system.status;
+
+import com.swirlds.common.wiring.component.InputWireLabel;
+import com.swirlds.platform.system.state.notifications.IssNotification;
+import com.swirlds.platform.system.status.actions.PlatformStatusAction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+
+/**
+ * A state machine that processes {@link PlatformStatusAction}s
+ */
+public interface StatusStateMachine {
+    /**
+     * Submit a status action
+     *
+     * @param action the action to submit
+     * @return the new status after processing the action, or null if the status did not change
+     */
+    @Nullable
+    @InputWireLabel("PlatformStatusAction")
+    PlatformStatus submitStatusAction(@NonNull final PlatformStatusAction action);
+
+    /**
+     * Process an ISS notification
+     * <p>
+     * This method will conditionally call {@link #submitStatusAction(PlatformStatusAction)}, depending on the
+     * type of ISS.
+     *
+     * @param issNotification the ISS notification
+     * @return the new status after processing the ISS, or null if the status did not change
+     */
+    @Nullable
+    @InputWireLabel("IssNotification")
+    PlatformStatus issOccurred(@NonNull final IssNotification issNotification);
```

Comment:
Out of curiosity, why was this method bubbled up to the top level API? Do you intend to eventually expose an input wire for each input type? Not necessarily a bad thing.

Reply:
This was since not all ISS notifications result in a status action.

As we discussed, refactoring so that this extra logic is a transformer on the `IssDetector`